src/curator/frontend_utils.py

```
@@ -0,0 +1,104 @@
+            # but will be invalid after the 'with' block closes.
+            print(f"Temporary file object created (will be auto-deleted on close). Path: {tmp_file.name}")
+            for chunk_num, chunk in enumerate(asset_response.iter_content(chunk_size=8192)):
+                if chunk_num % 10 == 0: # Log every 10 chunks
+                    print(f"Writing chunk {chunk_num} to temporary file...")
+                tmp_file.write(chunk)
+
+            tmp_file.flush() # Ensure all data is written to disk
+            tmp_file.seek(0) # Rewind the file to the beginning for reading by tarfile
+
+            print(f"Asset downloaded to temporary file. Checking if it's a valid tarfile...")
+            if not tarfile.is_tarfile(fileobj=tmp_file): # Check using the file object
```

Flake8 (7.2.0)

[error] at least two spaces before inline comment

(E261)